velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -156,7 +168,8 @@ struct UnixTimestampParseFunction {
       const std::vector<TypePtr>& /*inputTypes*/,
       const core::QueryConfig& config,
       const arg_type<Varchar>* /*input*/) {
-    format_ = buildJodaDateTimeFormatter(kDefaultFormat_);
+    format_ = getDateTimeFormatter(
+        config.sparkLegacyTimeParser(), kDefaultFormat_, false);
```

Comment:
How does Spark decide lenient or not? Is it through another configuration?

Reply:
There is no configuration to decide lenient or not. It's just uses lenient mode or strict mode . See this [Spark issue](https://issues.apache.org/jira/browse/SPARK-30788)

> SimpleDateFormat - is used in JDBC datasource, in partitions parsing.
SimpleDateFormat in strong mode (lenient = false). It is used by the date_format, from_unixtime, unix_timestamp and to_unix_timestamp functions.

Reply:
FYI:
Functions using lenient mode:
'from_unixtime',  'unix_timestamp', 'make_date', 'to_unix_timestamp', 'date_format'

Functions using strict mode(lenient=false):
cast date to string

Reply:
Sure. Updated.